linter/ignore.go

```diff
@@ -0,0 +1,98 @@
+package linter
+
+import (
+	"strings"
+
+	"github.com/ysugimoto/falco/ast"
+)
+
+// ignore signatures
+const (
+	falcoIgnoreNextLine = "falco-ignore-next-line"
+	falcoIgnoreThisLine = "falco-ignore"
+	falcoIgnoreStart    = "falco-ignore-start"
+	falcoIgnoreEnd      = "falco-ignore-end"
+)
+
+type ignore struct {
+	ignoreNextLine bool
+	ignoreThisLine bool
+	ignoreRange    bool
+}
+
+// Setup ignores for common statements, declarations.
+// For common statements:
+//
+// // leading comments
+// [STATEMENT] // trailing comments
+//
+// Then leading comments accept falco-ignore-next-line, falco-ignore-start, falco-ignore-end
+// trailing comments accept falco-ignore
+func (i *ignore) SetupStatement(meta *ast.Meta) {
+	// Find ignore signature in leading comments
+	for _, c := range meta.Leading {
+		line := strings.TrimLeft(c.String(), "#@*/ ")
+		switch {
+		case strings.HasPrefix(line, falcoIgnoreNextLine):
+			i.ignoreNextLine = true
+		case strings.HasPrefix(line, falcoIgnoreStart):
+			i.ignoreRange = true
+		case strings.HasPrefix(line, falcoIgnoreEnd):
+			i.ignoreRange = false
+		}
+	}
+
+	// Find ignore signature in trailing comments
+	for _, c := range meta.Trailing {
```

Comment:
What would happen in this case?

```
set req.http.foo = "bar"; // falco-ignore-start
set req.http.bar = underfined
//falco-ignore-end
```
The `falco-ignore-start` would be ignored right?

Reply:
Yes. the `// falco-ignore-start` expects that present in the leasing comment.
It means the trailing comment only accepts `// falco-ignore`, which is related to the current statement.

This comes from our parser logic:

```
set req.http.foo = "bar"; // falco-ignore-start  <- trailing comment
// falco-ignore-start <- leading comment
set req.http.foo = "bar";
```

So linefeed is important.